src/content_disposition.rs

```diff
@@ -0,0 +1,127 @@
+use percent_encoding::percent_decode;
+
+/// Parse filename from Content-Disposition header
+/// Prioritizes filename* parameter if present, otherwise uses filename parameter
+pub fn parse_filename_from_content_disposition(content_disposition: &str) -> Option<String> {
+    let parts: Vec<&str> = content_disposition
+        .split(';')
+        .map(|part| part.trim())
+        .collect();
+
+    // First try to find filename* parameter
+    for part in parts.iter() {
+        if part.starts_with("filename*=") {
```

Comment:
if you do it like this:
```suggestion
        if let Some(value) = part.strip_prefix("filename*=") {
```
then you don't need `trim_start_matches` in the function